File: Javascript/0101-0150/141. Linked List Cycle.js
// https://leetcode.com/problems/linked-list-cycle/

var hasCycle = function(head) {
    fast = head;
    slow = head;

    while(fast != null && fast.next != null){ //runs untill the fast reaches null
        slow = slow.next;
        fast = fast.next.next;
        if(fast == slow){//both meets so its a cycle
            return true;
        }
    }

    return false;
};

File: Python/0101-0150/141. Linked List Cycle.py
# https://leetcode.com/problems/linked-list-cycle/

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        fast = head
        slow = head

        while fast != None and fast.next != None: # runs untill the fast reaches null
            slow = slow.next
            fast = fast.next.next
            if fast == slow: # both meets so its a cycle
                return True

        return False
   
